play-json/shared/src/main/scala/JsMacroImpl.scala

```diff
@@ -25,16 +25,16 @@ import scala.reflect.macros.blackbox
     }
   }
 
-  def formatImpl[A: c.WeakTypeTag]: c.Expr[OFormat[A]] =
-    macroImpl[A, OFormat, Format](
+  def formatImpl[A: c.WeakTypeTag, O <: Json.MacroOptions]: c.Expr[OFormat[A]] =
```

Comment:
Goal was to be able to define both the compile and runtime config/options, there it's prevented by not being able to pass the implicit through macros.